Sofa/framework/Core/src/sofa/core/topology/BaseTopology.cpp

```diff
@@ -168,11 +169,11 @@ void TopologyContainer::resetTopologyHandlerList()
 {
     for (auto& topologyHandlerList : m_topologyHandlerListPerElement)
     {
-        for (auto it = topologyHandlerList.begin();
-            it != topologyHandlerList.end(); ++it)
+        std::for_each(topologyHandlerList.begin(), topologyHandlerList.end(), [](TopologyHandler* topoHandler) 
         {
-            *it = nullptr;
-        }
+            topoHandler = nullptr;
```

Comment:
I am not sure this is doing what you think. `topoHandler` is a copy, not a reference

Reply:
you are right.. 
And not sure the entire loop is really useful for this PR (I have more change incoming)